servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpMetaData.java

```diff
@@ -46,6 +47,30 @@
      */
     HttpHeaders headers();
 
+    /**
+     * The {@link ContentCoding} used to encode the payload {@link io.servicetalk.buffer.api.Buffer} of a
+     * request or a response.
+     * If the endpoint protocol is setup with {@link HttpProtocolConfig#supportedEncodings()} the server will
+     * auto-establish the accepted encoding for the response, unless the caller provides a specific encoding
+     * calling this method. Any encoding passed here, takes precedence. In other words, a compressed response, can
+     * be disabled by passing {@link ContentCodings#none()}.
+     *
+     * @param encoding The {@link ContentCoding} used for the encoding of the payload.
+     * @return {@code this}.
+     * @see <a href="https://tools.ietf.org/html/rfc7231#section-3.1.2.2">Content-Encoding</a>
+     */
+    HttpMetaData encoding(ContentCoding encoding);
```

Comment:
It's good that we know the answer how users can configure compression per-reqiest or per-response. They can specify different compression levels for different payloads. However, this is too advanced use-case. Wonder if we should start with a simple approach based on a filter and let users control it there. Let's discuss this offline.

Reply:
It is an advanced use-case indeed, and a filter would also work nicely (even though you would have to work with headers instead of using the API). The reason I kept this as part of the API is that compression/encoding is part of the vanilla HTTP spec, so it feels more natural to have an option to control that for a request/response, rather than having a proxy to do so. WDYT?